sympy/solvers/solveset.py

```diff
@@ -365,27 +398,72 @@ def _solve_as_rational(f, symbol, domain):
 
 def _solve_trig(f, symbol, domain):
     """ Helper to solve trigonometric equations """
-    f = trigsimp(f)
-    f_original = f
+    from sympy.polys import factor
+    from sympy.sets import ImageSet
+    if _is_function_class_equation(TrigonometricFunction, f, symbol):
+        f = trigsimp(f)
+    # trigsimp is not defined for hyperbolic
+    # TODO trigh function for Hyperbolic Functions
+    f_orig = f
     f = f.rewrite(exp)
-    f = together(f)
-    g, h = fraction(f)
-    y = Dummy('y')
-    g, h = g.expand(), h.expand()
-    g, h = g.subs(exp(I*symbol), y), h.subs(exp(I*symbol), y)
-    if g.has(symbol) or h.has(symbol):
-        return ConditionSet(symbol, Eq(f, 0), S.Reals)
-
-    solns = solveset_complex(g, y) - solveset_complex(h, y)
-
-    if isinstance(solns, FiniteSet):
-        result = Union(*[invert_complex(exp(I*symbol), s, symbol)[1]
-                       for s in solns])
-        return Intersection(result, domain)
-    elif solns is S.EmptySet:
-        return S.EmptySet
-    else:
-        return ConditionSet(symbol, Eq(f_original, 0), S.Reals)
+    soln = _solveset(f, symbol, S.Complexes)
+    if isinstance(soln,Union) and all(isinstance(s, ImageSet) for s in soln.args):
+        soln_2pi = []
+        soln_list = []
+        soln_len = len(soln.args)
+        for i in range(0,soln_len):
+            soln_list.append(soln.args[i])
+        for s in soln_list:
+            lamb = s.args[0]
+            val = lamb(0)
+            soln_2pi.append(val)
+        soln_2pi.sort()
+        a = Dummy('a', real=True)
+        b = Dummy('b', real=True)
+        n = Dummy('n', real =True)
+        equations = []
+        positive_eq = []
+        negative_eq = []
+        for j in range(0,len(soln_2pi)):
+            if soln_2pi[j]<0:
+                negative_eq.append(soln_2pi[j])
+            elif soln_2pi[j] > 0:
+                positive_eq.append(soln_2pi[j])
+        plen = len(positive_eq)
+        nlen = len(negative_eq)
+        if not (plen == 1 and nlen == 1):
+            positive_eq.sort()
+            negative_eq.sort()
+            negative_eq.reverse()
+            if not nlen == 1:
```

Comment:
After the discussion in [pull/10713](https://github.com/sympy/sympy/pull/10713#issuecomment-204390616) I found that instead of using these steps (linsolve/Dummy variables) we can use `interpolate` method defined in polys/polyfuncs.py